Lib/unittest/mock.py

```diff
@@ -1011,7 +1011,7 @@ def assert_has_calls(self, calls, any_order=False):
                 raise AssertionError(
                     f'{problem}\n'
                     f'Expected: {_CallList(calls)}'
-                    f'{self._calls_repr(prefix="  Actual").rstrip(".")}'
+                    f'{self._calls_repr(prefix="  Actual", display_empty=True).rstrip(".")}'
                 ) from cause
```

Comment:
`assert_called_with()` displays
```
  Actual: not called
```
if the mock was not called. I think that it is better to give here the same result for consistency.

Reply:
Sorry, I'd missed this comment before.

I don't think it matters too much, since you either have consistency with when there are calls with `assert_have_calls` (as it is now) or consistency with when there are no calls with `assert_called_with` (as it would be with "Actual: not called").

I think I slightly prefer "Actual: []", but I'd also definitely be happy to change it if you think "Actual: not called" makes more sense. Let me know what you think.

Reply:
I don't feel strongly, but happy to go with what's implemented.

Reply:
It is not a big deal, so keep what you prefer.

But note that it is the only one special call of `_calls_repr()`. It is only one with argument, and its result is postprocessed (removed a dot). And now you add yet one special parameter. Perhaps it would be simpler to inline its code.

Like this:
```py
                    f'Expected: {_CallList(calls)}\n'
                    f'  Actual: {safe_repr(self.mock_calls)}'
```

Reply:
True, that is nicer. I've applied that: https://github.com/python/cpython/pull/113573/commits/1c25be8fae8f81256ec1b80fbe5f524c9bf9571f